pkgs/by-name/so/sonarr/package.nix

```diff
@@ -0,0 +1,163 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, substituteAll
+, buildDotnetModule
+, dotnetCorePackages
+, sqlite
+, withFFmpeg ? true # replace bundled ffprobe binary with symlink to ffmpeg package.
+, ffmpeg
+, fetchYarnDeps
+, yarn
+, nodejs
+, nixosTests
+  # update script
+, writeShellScript
+, writers
+, python3Packages
+, nix
+, prefetch-yarn-deps
+}:
+let
+  ffprobe = lib.getExe' ffmpeg "ffprobe";
+in
+buildDotnetModule rec {
+  pname = "sonarr";
+  version = "4.0.1.929";
+
+  src = fetchFromGitHub {
+    owner = "Sonarr";
+    repo = "Sonarr";
+    rev = "v${version}";
+    hash = "sha256-TOUCJtQKteDfeSgfxJ2xdJch0RlCcO5YjoqPUvOhcRY=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./buildprops.patch;
+      copyrightYear = "2024";
+      assemblyVersion = version;
+      assemblyConfiguration = "main";
+    })
+  ];
+
+  strictDeps = true;
+  nativeBuildInputs = [ nodejs yarn prefetch-yarn-deps ];
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-N/yeKkYTpOTkwl2o9YjfioeBxUf3YSZ692nuQcb7PzM=";
+  };
+
+  postConfigure = ''
+    yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+    fixup-yarn-lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules
+  '';
+  postBuild = ''
+    yarn --offline run build --env production
+  '';
+  postInstall = lib.optionalString (withFFmpeg) ''
+    rm $out/lib/sonarr/ffprobe
+    ln -s ${lib.escapeShellArg ffprobe} $out/lib/sonarr/ffprobe
+  '' + ''
+    cp -a _output/UI $out/lib/sonarr/UI
+  '';
+  # Add an alias for compatibility with Sonarr v3 package.
+  postFixup = ''
+    ln -s Sonarr $out/bin/NzbDrone
+  '';
+
+  nugetDeps = ./deps.nix;
+  nugetConfigFile = "src/NuGet.Config";
+
+  runtimeDeps = [ sqlite ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-runtime = dotnetCorePackages.aspnetcore_6_0;
+
+  # Nixpkgs .NET build infrastructure uses globalization-invariant mode that
+  # breaks a lot of tests when run on Darwin. Instead of trying to disable them,
+  # just do not run all tests. See https://github.com/NixOS/nixpkgs/pull/217587
+  doCheck = !stdenv.hostPlatform.isDarwin;
+
+  executables = [ "Sonarr" ];
```

Comment:
In the previous version the executable was (rather unfortunately) called `NzbDrone`, and the systemd unit refers to that too IIRC, so you might need to update that too.

Reply:
Done. Though we also create NzbDrone symlink in postFixup hook for compatibility.